evalml/tests/automl_tests/test_automl_utils.py

```diff
@@ -282,3 +286,26 @@ def test_get_best_sampler_for_data_sampler_method(
             assert name_output == "SMOTENC Oversampler"
         else:
             assert name_output == "SMOTEN Oversampler"
+
+
+def test_get_hyperparameter_ranges():
+    pipeline_ = BinaryClassificationPipeline(
+        component_graph=["Imputer", "Random Forest Classifier"]
+    )
+    custom_hyperparameters_ = {
+        "Imputer": {"numeric_impute_strategy": Categorical(["most_frequent", "mean"])},
+        "Random Forest Classifier": {"n_estimators": Integer(150, 160)},
+    }
+    algo = IterativeAlgorithm(
+        allowed_pipelines=[pipeline_],
+        random_seed=0,
+        custom_hyperparameters=custom_hyperparameters_,
+    )
+    algo_ranges = algo._tuners[
+        "Random Forest Classifier w/ Imputer"
+    ]._pipeline_hyperparameter_ranges
+    hyper_ranges = get_hyperparameter_ranges(
+        pipeline_.component_graph, custom_hyperparameters_
+    )
+
+    assert algo_ranges == hyper_ranges
```

Comment:
Great call adding direct coverage of `get_hyperparameter_ranges`, we definitely need that now that a bunch of the old tests from `test_pipelines.py` are deleted!

This test currently checks that the output of `get_hyperparameter_ranges` matches the ranges used inside the automl algo (private variable). I think it would be more useful if it instead checked that the output of `get_hyperparameter_ranges` matches a hard-coded dictionary or something. You have other (great) behavioral tests in `test_automl.py` and the iterative algo tests which provide coverage of the algo behavior, so IMO there's no need to create an `IterativeAlgorithm` instance here and check a private field.

Reply:
Maybe we can recycle some of the deleted pipeline tests that checked that the pipeline `custom_hyperparameters` attribute was set correctly?